File: SW/D2_1940.java
import java.util.*;

public class D2_1940 {
    static int N, res, s;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        for (int t=1; t<=T; t++) {
            N = sc.nextInt();
            sc.nextLine();

            res = 0;
            s = 0;
            for (int i=0; i<N; i++) {
                String[] comm = sc.nextLine().split(" ");
                if (comm[0].equals("0")) {
                    res += s;
                } else if (comm[0].equals("1")) {
                    s += Integer.parseInt(comm[1]);
                    res += s;
                } else {
                    if (s > Integer.parseInt(comm[1])) {
                        s -= Integer.parseInt(comm[1]);
                        res += s;
                    } else {
                        s = 0;
                    }
                }
            }
            System.out.printf("#%d %d\n", t, res);
        }
    }    
}

File: SW/D2_1940.py
for t in range(1, int(input())+1):
    N = int(input())
    res, speed = 0, 0

    for _ in range(N):
        v = list(map(int, input().split()))
        if v[0] == 0:
            res += speed
        elif v[0] == 1:
            speed += v[1]
            res += speed
        else:
            speed = speed-v[1] if speed > v[1] else 0
            res += speed
        
    print("#{} {}".format(t, res))